src/radical/pilot/session.py

```diff
@@ -285,11 +285,15 @@ def close(self, cleanup=True, terminate=True, delete=None):
             logger.error("Session object already closed.")
             return
 
+        # set defaults
+        if cleanup   == None: cleanup   = True
+        if terminate == None: terminate = True
+
```

Comment:
It's not solving any problem - I find it easier to read.  That is the way python needs default parameter values for mutable types, and I find it easier to use the same way for all types really, then one way for one type class, another way for another type class (specifically since python does not even define the types, and we have many places where we pass either an instance or a list).